### PR TITLE
feat: Add tracking + WorksFollowed rail

### PR DESCRIPTION
**Pull request by damassi**
**Addresses**: 
- https://artsyproduct.atlassian.net/browse/GRO-197
- https://artsyproduct.atlassian.net/browse/GRO-233

**Analytics Doc**:
- https://docs.google.com/spreadsheets/d/12wj1rt-j5SguboPJMuaM1GqJD-5B5u5piipw8MQ5Zuk/edit#gid=243169971

This adds the new `WorksByArtistsYouFollow` rail and wires up tracking. 

There's still a couple open tracking qs in slack but that can be addressed in follow-up PR since its based around possible new functionality. 

Will also add follow-up PR with analytics tests once https://github.com/artsy/force/pull/7272 is merged. 

cc @artsy/grow-devs 

works

### CODE REVIEW

**Review by damassi — COMMENTED** [review]
Adding some notes for you @The-Beez-Kneez. Tracking is like relay, there's some conventions that one has to learn and it can basically only happen by diving in and just getting experience... Fear not ⚔️

File: src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx [hunk 1]
```diff
@@ -5,10 +5,26 @@ import { AuctionArtworksRail_sale } from "v2/__generated__/AuctionArtworksRail_s
...
 import { AuctionArtworksRailArtworksQueryRenderer } from "./AuctionArtworksRailArtworks"
-import { AuctionArtworksRailPlaceholder } from "./AuctionArtworksRailPlaceholder"
+import { AuctionArtworksRailPlaceholder } from "../AuctionArtworksRailPlaceholder"
+import { getContextModule } from "../../Utils/getContextModule"
+import { useTracking } from "react-tracking"
+import {
+  clickedArtworkGroupHeader,
+  ClickedArtworkGroupHeaderArgs,
+  OwnerType,
+} from "@artsy/cohesion"
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx:15]: Cohesion is where we store our analytics schema: https://github.com/artsy/cohesion. When you get a sec give the readme and [the docs](https://cohesion.artsy.net/) a good read just to get kinda familiar.

File: src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx [hunk 2]
```diff
@@ -21,10 +37,28 @@ export const AUCTION_ARTWORKS_IMAGE_HEIGHT = 160
...
 }) => {
   const ref = useRef<HTMLDivElement | null>(null)
+  const { trackEvent } = useTracking()
   const { isEnteredView, Waypoint } = useLazyLoadComponent()
+  const { contextPageOwnerType } = useAnalyticsContext()
+  const contextModule = getContextModule(tabType)
+
+  const trackViewSaleClick = () => {
+    trackEvent(
+      clickedArtworkGroupHeader({
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx:51]: For lots and lots of common interactions there's helper functions that one can import and fill out; here's an example of possible [click helpers](https://github.com/artsy/cohesion/tree/master/src/Helpers/Click), though there's lots of stuff in that parent `/Helpers` directory.

File: src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx [hunk 3]
```diff
@@ -21,10 +37,28 @@ export const AUCTION_ARTWORKS_IMAGE_HEIGHT = 160
 
 export const AuctionArtworksRail: React.FC<AuctionArtworksRailProps> = ({
   sale,
+  tabType,
   ...rest
 }) => {
   const ref = useRef<HTMLDivElement | null>(null)
+  const { trackEvent } = useTracking()
   const { isEnteredView, Waypoint } = useLazyLoadComponent()
+  const { contextPageOwnerType } = useAnalyticsContext()
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/AuctionArtworksRail/AuctionArtworksRail.tsx:46]: Generally speaking, every tracking event needs this `contextPageOwnerType`, which can be pulled from our handy hook.

File: src/v2/Apps/Auctions/Components/MyBids/MyBidsBidHeader.tsx [hunk 4]
```diff
@@ -3,14 +3,33 @@ import { createFragmentContainer, graphql } from "react-relay"
...
+      to={`/auction/${sale.slug}`}
+      noUnderline
+      onClick={() => {
+        trackEvent(
+          clickedEntityGroup({
+            contextModule: ContextModule.yourActiveBids,
+            contextPageOwnerType,
+            destinationPageOwnerType: OwnerType.sale,
+            type: "thumbnail",
+          })
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/MyBids/MyBidsBidHeader.tsx:29]: Another example of a tracking helper function

File: src/v2/Apps/Auctions/Components/MyBids/MyBidsBidItem.tsx [hunk 5]
```diff
@@ -13,16 +13,40 @@ import {
...
   saleArtwork: MyBidsBidItem_saleArtwork
 }
 
 export const MyBidsBidItem: React.FC<MyBidsBidItemProps> = ({
+  horizontalSlidePosition,
   saleArtwork,
 }) => {
+  const { trackEvent } = useTracking()
+  const { contextPageOwnerType } = useAnalyticsContext()
+  const contextModule = getContextModule("myBids")
```
Comment (jonallured) [inline src/v2/Apps/Auctions/Components/MyBids/MyBidsBidItem.tsx:32]: This will always be `ContextModule.yourActiveBids`, no?

**Review by jonallured — COMMENTED** [review]
Love to see how quickly that works for you rail came together!! 😍 

I had some comments about tracking and the context module. There's a function called `getContextModule` that seems of dubious utility to me. There are a couple uses that seem to be overkill - the "my bids" and "works for you" rails will always return a certain context module. I think this abstraction can be removed or am I missing something? 🤔

File: src/v2/Apps/Auctions/Utils/getContextModule.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { ContextModule } from "@artsy/cohesion"
+import { TabType } from "../Components/AuctionArtworksRail/AuctionArtworksRail"
+
+/**
+ * Returns the context module for tracking purposes
+ */
+export function getContextModule(tabType: TabType) {
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/v2/Apps/Auctions/Utils/getContextModule.ts:7]: Yah this function never sat right with me either -- the reason its needed tho is we share the raiils across a few different types of tabs, so we need to compute the context and then pass it in.

Will update to an map tho 👍

**Pull request merged**